cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/impl/PdfReportStructure.java

```diff
@@ -52,30 +52,32 @@ public void addStaticElementsForEachPage(ReportElementStatic... elements) {
     public void expandPagesStaticElements(int totalPages) {
         for (int i = staticElementsForEachPage.size() - 1; i >= 0; --i) {
             ReportElementStatic elem = staticElementsForEachPage.get(i);
-            ReportElement baseElement = elem.getBase();
             for (int pageNo = 0; pageNo < totalPages; pageNo++) {
-                setFooterPageNumbers(baseElement, pageNo, totalPages);
-                if (baseElement instanceof ReportTable) {
-                    ReportTable baseTable = (ReportTable) baseElement;
-                    for (int row = baseTable.getElements().length - 1; row >= 0 ; --row){
-                        for (int col = baseTable.getElements()[row].length - 1; col >= 0; --col) {
-                            setFooterPageNumbers(baseTable.getElements()[row][col], pageNo, totalPages);
-                        }
-                    }
-                }
-                if ((elem.getFooterOnPages() == ReportFooterOnPages.ALL) ||
-                        (elem.getFooterOnPages() == ReportFooterOnPages.ALL_BUT_FIRST && pageNo != 0)
-                        || (elem.getFooterOnPages() == ReportFooterOnPages.ALL_BUT_LAST && pageNo != (totalPages - 1)))
-                    addStaticElement(new ReportElementStatic(baseElement, pageNo, elem.getX(), elem.getY(), elem.getWidth(), elem.getFooterOnPages()));
+                if ((elem.getFooterOnPages() == ReportFooterOnPages.ALL) || (elem.getFooterOnPages() == ReportFooterOnPages.ALL_BUT_FIRST && pageNo != 0) ||
+                        (elem.getFooterOnPages() == ReportFooterOnPages.ALL_BUT_LAST && pageNo != (totalPages - 1)))
+                    addStaticElement(new ReportElementStatic(getFooterElementWithPagesSet(elem.getBase(), pageNo, totalPages), pageNo, elem.getX(), elem.getY(), elem.getWidth(), elem.getFooterOnPages()));
```

Comment:
this could be cleanup up:

```java
ReportFooterOnPages config = elem.getFooterOnPages();
if (pageNo == 0 && 
    config == ReportFooterPages.ALL_BUT_FIRST) {
  continue;
} else if (pageNo == (totalPages - 1) && 
           config == ReportFooterPages.ALL_BUT_LAST) {
 continue;
}

addStaticElement(
    new ReportElementStatic(getFooterElementWithPagesSet(elem.getBase(), pageNo, totalPages),
        pageNo,
        elem.getX(),
        elem.getY(),
        elem.getWidth(),
        elem.getFooterOnPages())
);
```

of course it can be written via or concatenation but it gets much harder to read